File: back-end-maven/src/main/java/net/weg/wegssm/model/service/BuService.java
package net.weg.wegssm.model.service;

import lombok.AllArgsConstructor;
import net.weg.wegssm.model.entities.Bu;
import net.weg.wegssm.repository.BuRepository;
import org.springframework.stereotype.Service;

import java.util.List;

/**
 * Classe service para as buscas de BUs
 */
@Service
@AllArgsConstructor
public class BuService {

    /**
     * Classe repository das BUs
     */
    private BuRepository buRepository;

    /**
     * Função para buscar todas as BUs salvas no banco
     * @return
     */
    public List<Bu> findAll() {
        return buRepository.findAll();
    }

}

File: front-end/src/service/buService.js
import axios from "./api";

/** Classe service para as BUs */
class BuService {

    /** Função para buscar todas as BUs salvas no banco */
    async getAll() {
        return (await axios.get(`/bu`, { withCredentials: true })).data;
    }
}

export default new BuService();